pkg/image/prune/prune.go

```diff
@@ -671,71 +684,95 @@ func calculatePrunableImageComponents(g graph.Graph) []*imagegraph.ImageComponen
 	return components
 }
 
+func getPrunableComponents(g graph.Graph, prunableImageIDs graph.NodeSet) []*imagegraph.ImageComponentNode {
+	graphWithoutPrunableImages := subgraphWithoutPrunableImages(g, prunableImageIDs)
+	return calculatePrunableImageComponents(graphWithoutPrunableImages)
+}
+
 // pruneStreams removes references from all image streams' status.tags entries
 // to prunable images, invoking streamPruner.UpdateImageStream for each updated
 // stream.
-func pruneStreams(g graph.Graph, imageNodes []*imagegraph.ImageNode, streamPruner ImageStreamDeleter) []error {
-	errs := []error{}
+func pruneStreams(
+	g graph.Graph,
+	prunableImageNodes map[string]*imagegraph.ImageNode,
+	streamPruner ImageStreamDeleter,
+) error {
+	prunableStreams := make(map[string]*imagegraph.ImageStreamNode)
 
 	glog.V(4).Infof("Removing pruned image references from streams")
-	for _, imageNode := range imageNodes {
+	for _, imageNode := range prunableImageNodes {
 		for _, n := range g.To(imageNode) {
 			streamNode, ok := n.(*imagegraph.ImageStreamNode)
 			if !ok {
 				continue
 			}
 
-			if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-				stream, err := streamPruner.GetImageStream(streamNode.ImageStream)
-				if err != nil {
-					return err
-				}
-				updatedTags := sets.NewString()
-
-				glog.V(4).Infof("Checking if ImageStream %s has references to image %s in status.tags", getName(stream), imageNode.Image.Name)
+			streamName := getName(streamNode.ImageStream)
+			prunableStreams[streamName] = streamNode
+		}
+	}
 
-				for tag, history := range stream.Status.Tags {
-					glog.V(4).Infof("Checking tag %q", tag)
+	for streamName, streamNode := range prunableStreams {
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			stream, err := streamPruner.GetImageStream(streamNode.ImageStream)
+			if err != nil {
+				// During prune someone removed imagestream
+				if kerrapi.IsNotFound(err) {
+					return nil
```

Comment:
*nit*: I think it's worth of 4th info level at least.

Reply:
@miminar Done